dataplug-twitter/app/org/hatdex/dataplugTwitter/apiInterfaces/TwitterFollowerInterface.scala

```diff
@@ -41,32 +41,102 @@ class TwitterFollowerInterface @Inject() (
       case "0" => None
+
       case cursor: String => {
-        Some(params.copy(queryParameters = params.queryParameters + ("cursor" -> cursor)))
+        logger.debug("Cursor found")
+        val tempParams = checkIfWeHaveNewFollowers(content, checkForMostRecentFollower(content, params))
+        tempParams match {
+          case Some(parameters) =>
+            val maybeMostRecentFollower = params.storageParameters.flatMap(_.get("mostRecentFollower"))
+            maybeMostRecentFollower match {
+              case Some(_) => Some(parameters.copy(queryParameters = parameters.queryParameters + ("cursor" -> "-1")))
+              case _       => Some(parameters.copy(queryParameters = parameters.queryParameters + ("cursor" -> cursor)))
+            }
+
+          case _ =>
+            logger.debug(s"No new parameters")
+            None
+        }
       }
     }
   }
 
   def buildNextSync(content: JsValue, params: ApiEndpointCall): ApiEndpointCall = {
-    //implement
-    //    val maybeNextSyncToken = (content \ "nextSyncToken").asOpt[String]
-    //
-    //    maybeNextSyncToken map { nextSyncToken =>
-    //      val updatedParams = params.queryParameters + ("syncToken" -> nextSyncToken) - "pageToken"
-    //      params.copy(queryParameters = updatedParams)
-    //    } getOrElse {
-    //      params
-    //    }
-    params
+    updateStorageParametersMostRecentFollower(content, params)
+  }
+
+  private def checkForMostRecentFollower(content: JsValue, params: ApiEndpointCall): ApiEndpointCall = {
+    val maybeMostRecentFollower = params.storageParameters.flatMap(_.get("tempMostRecentFollower"))
+    maybeMostRecentFollower match {
+      case None =>
+        (content \ "users").as[JsArray].value.head.validate[TwitterUser].map { user =>
+          logger.debug(s"the most recent follower found in JSON is:  ${{ user.id.toString }}")
+          params.copy(storageParameters = Some(params.storage ++ Map("tempMostRecentFollower" -> user.id.toString)))
+        }.getOrElse(params)
+
+      case _ => params
+    }
+  }
+
+  private def checkIfWeHaveNewFollowers(content: JsValue, params: ApiEndpointCall): Option[ApiEndpointCall] = {
+    val maybeTempMostRecentFollower = params.storageParameters.flatMap(_.get("tempMostRecentFollower"))
+    val maybeMostRecentFollower = params.storageParameters.flatMap(_.get("mostRecentFollower"))
+
+    logger.debug(s"maybeTempMostRecentFollower $maybeTempMostRecentFollower")
+    logger.debug(s"maybeMostRecentFollower $maybeMostRecentFollower")
+
+    maybeMostRecentFollower match {
+      case Some(value) => // This is NOT the first sync ever
+
+        maybeTempMostRecentFollower match {
+          case Some(_) => //means this is not the first sync EVER and we have gone through the new data
+            logger.debug("means this is not the first sync EVER and we have gone through the new data")
+            val users = (content \ "users").as[JsArray].validate[Seq[TwitterUser]].get
```

Comment:
- failure to validate `[Seq[TwitterUser]]` unhandled
- `as[JsArray]` unnecessary